### PR TITLE
Adding Manifesto to README

### PR DESCRIPTION
**Pull request by vincentqb**
We want to describe 
* the standardization
* the philosophy behind the behavior of transformations.

See also
* [comment in #166](https://github.com/pytorch/audio/pull/166#issuecomment-515603852)
* [comment in keunwoochoi/torchaudio-contrib#36](https://github.com/keunwoochoi/torchaudio-contrib/issues/36#issuecomment-502144987)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,39 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+* Spectrogram: (channel, time) -> (channel, frequency, time, 2)
+* MelScale: (channel, time) -> (channel, n_mels, time)
+* MFCC: (channel, time) -> (channel, n_mfcc, time)
+* MuLawEncode: (channel, time) -> (channel, n_mulaw, time)
+* MuLawDecode: (channel, n_mulaw, time) -> (channel, time)
+* Resample: (channel, time) -> (channel, time)
+* STFT: (channel, time) -> (channel, frequency, time, 2).
+* ISTFT: (channel, frequency, time) -> (channel, time, 2).
+
+A spectrogram can be converted to DB scale or Mel scale, using AmplitudeToDB and AmplitudeToMel.
```
Comment (vincentqb) [inline README.md]: We do not have that yet.

File: README.md [hunk 2]
```diff
@@ -54,6 +54,39 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+Transforms expect the following shapes. In particular, the input of all transforms and functions assumes channel first.
+
+* Spectrogram: (channel, time) -> (channel, frequency, time, 2)
+* MelScale: (channel, time) -> (channel, n_mels, time)
+* MFCC: (channel, time) -> (channel, n_mfcc, time)
+* MuLawEncode: (channel, time) -> (channel, n_mulaw, time)
+* MuLawDecode: (channel, n_mulaw, time) -> (channel, time)
+* Resample: (channel, time) -> (channel, time)
+* STFT: (channel, time) -> (channel, frequency, time, 2).
+* ISTFT: (channel, frequency, time) -> (channel, time, 2).
```
Comment (vincentqb) [inline README.md]: @jamarshon -- ISTFT takes (channel, frequency, time, 2)? Does STFT do the same?

**Comment by vincentqb** [comment]
@cpuhrsch @jamarshon -- thoughts? is there anything else i should make sure to cover?

File: README.md (hunk 2 above)
Comment (jamarshon) [reply README.md]: STFT: 
(channel, time) -> (channel, frequency, time, 2) or
(time) -> ( frequency, time, 2) 
ISTFT:
(channel, frequency, time, 2) -> (channel, time) or
(frequency, time, 2)  -> (time)
so ISTFT is the inverse of STFT

Comment (vincentqb) [reply README.md]: Thanks for clarifying. I've updated.

File: README.md (hunk 1 above)
Comment (jamarshon) [reply README.md]: Going to add AmplitudeToDB but let's have AmplitudeToMel in another release

File: README.md [hunk 3]
```diff
@@ -54,6 +54,39 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+* n_mfcc, n_mels: to be consistent with other similarly named variables, with shape (channel, n_mfcc, time) and (channel, n_mels, times)
+* win_length: the length of the STFT window
+* window_fn: for functions that creates windows e.g. torch.hann_window
+
+Transforms expect the following shapes. In particular, the input of all transforms and functions assumes channel first.
+
+* Spectrogram: (channel, time) -> (channel, frequency, time, 2)
+* MelScale: (channel, time) -> (channel, n_mels, time)
+* MFCC: (channel, time) -> (channel, n_mfcc, time)
+* MuLawEncode: (channel, time) -> (channel, n_mulaw, time)
```
Comment (jamarshon) [inline README.md]: MuLawEncode/Decode doesn't change the shape of the tensor does it?
```
>>> torchaudio.functional.mu_law_encoding(torch.rand((2,10)), 3).shape
torch.Size([2, 10])
```

File: README.md [hunk 4]
```diff
@@ -54,6 +54,39 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+* MuLawDecode: (channel, n_mulaw, time) -> (channel, time)
+* Resample: (channel, time) -> (channel, time)
+* STFT: (channel, time, 2) -> (channel, frequency, time, 2).
+* ISTFT: (channel, frequency, time, 2) -> (channel, time, 2).
+
+A spectrogram can be converted to DB scale or Mel scale, using AmplitudeToDB and AmplitudeToMel.
+
+The Kaldi compliance interface follows Kaldi's interface, and provides access to: Kaldi's `fbank`, `spectrogram`, and `resample_waveform`.
```
Comment (jamarshon) [inline README.md]: Not sure if this is part of conventions. I would not include it as it is mentioned here https://github.com/pytorch/audio/pull/166/files#diff-04c6e90faac2675aa89e2176d2eec7d8R15

File: README.md (hunk 3 above)
Comment (vincentqb) [reply README.md]: Thanks for pointing this out!

File: README.md (hunk 4 above)
Comment (vincentqb) [reply README.md]: I agree with you. The Kaldi interface is to help onboard, but not really part of the conventions.

File: README.md (hunk 1 above)
Comment (vincentqb) [reply README.md]: I agree :)

File: README.md [hunk 5]
```diff
@@ -54,6 +54,36 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+
+* waveform: a tensor of audio samples with shape (channels, time)
+* sample_rate: the rate of audio samples (samples per second)
+* specgram: a tensor of spectrogram with shape (channels, time)
+* mel_specgram: a mel spectrogram with shape (channels, time)
+* hop_length: the number of samples between the starts of consecutive frames
+* n_freqs: the number of bins in a linear spectrogram
+* min_freq: the lowest frequency of the lowest band in a spectrogram
+* max_freq: the highest frequency of the highest band in a spectrogram
+* n_fft: the number of fourier bins
```
Comment (jamarshon) [inline README.md]: fourier->Fourier

File: README.md [hunk 6]
```diff
@@ -54,6 +54,36 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+* n_mfcc, n_mels: to be consistent with other similarly named variables, with shape (channel, n_mfcc, time) and (channel, n_mels, times)
+* win_length: the length of the STFT window
+* window_fn: for functions that creates windows e.g. torch.hann_window
+
+Transforms expect the following shapes. In particular, the input of all transforms and functions assumes channel first.
+
+* Spectrogram: (channel, time) -> (channel, frequency, time, 2)
+* AmplitudeToDB: (channel, frequency, time, 2) -> (channel, frequency, time, 2)
+* MelScale: (channel, time) -> (channel, n_mels, time)
```
Comment (jamarshon) [inline README.md]: Add MelSpectrogram maybe

File: README.md [hunk 7]
```diff
@@ -54,6 +54,36 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
 API Reference is located here: http://pytorch.org/audio/
+
+Conventions
+-----------
+
+Torchaudio is standardized around the following naming conventions.
+
+* waveform: a tensor of audio samples with shape (channels, time)
+* sample_rate: the rate of audio samples (samples per second)
+* specgram: a tensor of spectrogram with shape (channels, time)
```
Comment (jamarshon) [inline README.md]: (channels, frequency, time)

File: README.md [hunk 8]
```diff
@@ -54,6 +54,36 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+* mel_specgram: a mel spectrogram with shape (channels, time)
```
Comment (jamarshon) [inline README.md]: (channels, frequency, time)

File: README.md (hunk 5 above)
Comment (vincentqb) [reply README.md]: good catch!

File: README.md (hunk 6 above)
Comment (vincentqb) [reply README.md]: I agree. Also, changed frequency to n_freqs to align with n_mels here.

File: README.md [hunk 9]
```diff
@@ -54,6 +54,37 @@ torchaudio.save('foo_save.mp3', sound, sample_rate) # saves tensor to file
...
+
+Torchaudio is standardized around the following naming conventions.
+
+* waveform: a tensor of audio samples with dimensions (channel, time)
+* sample_rate: the rate of audio dimensions (samples per second)
+* specgram: a tensor of spectrogram with dimensions (channel, freq, time)
+* mel_specgram: a mel spectrogram with dimensions (channel, mel, time)
+* hop_length: the number of samples between the starts of consecutive frames
+* n_fft: the number of Fourier bins
+* n_mfcc, n_mel: the number of mel and MFCC bins
```
Comment (jamarshon) [inline README.md]: the order of this should be consistent:
n_mfcc, n_mel: the number of MFCC and mel bins

Comment (vincentqb) [reply README.md]: yes, thanks!

**Pull request merged**